bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/dto/SurePetcareDevice.java

```diff
@@ -44,7 +45,8 @@ private ProductType(int id, String name) {
         }
 
         public static @NonNull ProductType findByTypeId(final int id) {
```

Comment:
Maybe it's easy to annotate the class and its super class with `@NonNullByDefault` so `@NonNull` can be removed?

Reply:
I prefer to leave it as is. Changing this to `@NonNullByDefault` would also mean i have to look after all other members without knowing anything from this binding. Maybe something to revisit in the future.